network/peer.go

```diff
@@ -50,8 +60,9 @@ type peer struct {
 
 	// ip may or may not be set when the peer is first started. is only modified
 	// on the connection's reader routine.
-	ip     utils.IPDesc
-	ipLock sync.RWMutex
+	ip      utils.IPDesc
+	aliases []alias
```

Comment:
// aliases is a list of IPs other than [ip] that we have connected to this peer at.